util/rate_limiter.cc

```diff
@@ -138,98 +141,150 @@ void GenericRateLimiter::Request(int64_t bytes, const Env::IOPriority pri,
 
   do {
     bool timedout = false;
-    // Leader election, candidates can be:
-    // (1) a new incoming request,
-    // (2) a previous leader, whose quota has not been not assigned yet due
-    //     to lower priority
-    // (3) a previous waiter at the front of queue, who got notified by
-    //     previous leader
+
+    // Leader election:
+    //  Leader request's duty:
+    //  (1) Waiting for the next refill time;
+    //  (2) Refilling the bytes and granting requests.
```

Comment:
These two lines of comments were originally created by @ajkr in https://github.com/facebook/rocksdb/pull/8602/files#r682014946. They are merged into this PR only because they align with this PR's goal of improving rate limiter's readability with minor code changes and can serve as a good follow-up improvement of this PR.